components/server/src/server.ts

```diff
@@ -209,7 +211,7 @@ export class Server<C extends GitpodClient, S extends GitpodServer> {
                 session: req.session
             });
             if (!this.isAnsweredRequest(req, response)) {
-                response.status(500).send({ error: msg });
+                response.status(status).send({ error: msg });
```

Comment:
@csweichel @geropl Do you remember why we hide original error codes? Particularly fro code-sync endpoint it is very important that VS Code see them to show proper UI otherwise it fails silently.

What would be the proper fix? Just unwrap codes from code sync or like that is fine as well?

Reply:
I'd reckon we just did that out of habit. We've never really forwarded those status codes and just didn't do it here either.